ts/features/bonus/cgn/screens/merchants/CgnMerchantCategoriesListScreen.tsx

```diff
@@ -9,27 +8,28 @@ import {
   ListItemAction,
   ListItemNav
 } from "@pagopa/io-app-design-system";
-import * as React from "react";
-import { FlatList, RefreshControl, View } from "react-native";
+import * as pot from "@pagopa/ts-commons/lib/pot";
+import { useNavigation } from "@react-navigation/native";
 import { pipe } from "fp-ts/lib/function";
 import * as O from "fp-ts/lib/Option";
-import { useNavigation } from "@react-navigation/native";
+import * as React from "react";
+import { FlatList, RefreshControl, View } from "react-native";
 import { useSafeAreaInsets } from "react-native-safe-area-context";
 import { ProductCategoryWithNewDiscountsCount } from "../../../../../../definitions/cgn/merchants/ProductCategoryWithNewDiscountsCount";
-import { getCategorySpecs } from "../../utils/filters";
 import I18n from "../../../../../i18n";
 import { IOStackNavigationProp } from "../../../../../navigation/params/AppParamsList";
-import { CgnDetailsParamsList } from "../../navigation/params";
 import { useIODispatch, useIOSelector } from "../../../../../store/hooks";
-import { cgnCategoriesListSelector } from "../../store/reducers/categories";
-import { cgnCategories } from "../../store/actions/categories";
-import CGN_ROUTES from "../../navigation/routes";
-import { useIOBottomSheetAutoresizableModal } from "../../../../../utils/hooks/bottomSheet";
 import { isDesignSystemEnabledSelector } from "../../../../../store/reducers/persistedPreferences";
+import { useIOBottomSheetAutoresizableModal } from "../../../../../utils/hooks/bottomSheet";
+import { CgnDetailsParamsList } from "../../navigation/params";
+import CGN_ROUTES from "../../navigation/routes";
+import { cgnCategories } from "../../store/actions/categories";
+import { cgnCategoriesListSelector } from "../../store/reducers/categories";
+import { getCategorySpecs } from "../../utils/filters";
 
 export const CgnMerchantCategoriesListScreen = () => {
   const insets = useSafeAreaInsets();
-  const isFirstRender = React.useRef<boolean>(true);
+  const [isFirstRender, setIsFirstRender] = React.useState<boolean>(true);
```

Comment:
Instead saving the state of the first render, since we want that the "refreshing indicator" will appears only when the user pulls down the list, what do you think about saving the state of `isRefreshing` and set it as default value to `false`?

```suggestion
  const [isRefreshing, setIsRefreshing] = React.useState<boolean>(false);
```